examples/api/src/app/api/livepeer-video/route.ts

```diff
@@ -2,92 +2,86 @@ import { NextRequest, NextResponse } from "next/server";
 
 export async function POST(request: NextRequest) {
   const form = await request.formData();
-  // https://docs.livepeer.org/reference/api#upload-an-asset
-  const requestedUrlReq = await fetch(
-    "https://livepeer.studio/api/asset/request-upload",
+
+  const controller = new AbortController();
+  const signal = controller.signal;
+
+  // Cancel upload if it takes longer than 15s
+  setTimeout(() => {
+    controller.abort();
+  }, 15_000);
+
+  const uploadRes: Response | null = await fetch(
+    "https://ipfs.infura.io:5001/api/v0/add",
     {
       method: "POST",
+      body: form,
       headers: {
-        Authorization: `Bearer ${process.env.LIVEPEER_API_SECRET}`,
-        "Content-Type": "application/json",
+        Authorization:
+          "Basic " +
+          Buffer.from(
+            process.env.INFURA_API_KEY + ":" + process.env.INFURA_API_SECRET
+          ).toString("base64"),
       },
-      body: JSON.stringify({
-        name: "video.mp4",
-        staticMp4: true,
-        playbackPolicy: {
-          type: "public",
-        },
-        storage: {
-          ipfs: true,
-        },
-      }),
+      signal,
     }
   );
 
-  const requestedUrl = await requestedUrlReq.json();
+  const { Hash: hash } = await uploadRes.json();
 
-  const url = requestedUrl.url;
+  const responseData = { url: `ipfs://${hash}` };
 
-  const videoUpload = await fetch(url, {
-    method: "PUT",
-    headers: {
-      Authorization: `Bearer ${process.env.LIVEPEER_API_SECRET}`,
-      "Content-Type": "video/mp4",
-    },
-    body: form.get("file"),
-  });
+  return NextResponse.json({ data: responseData });
+}
 
-  if (videoUpload.status >= 400) {
-    return NextResponse.json(
-      { message: "Something went wrong" },
-      {
-        status: videoUpload.status,
-      }
-    );
-  }
+// needed for preflight requests to succeed
+export const OPTIONS = async (request: NextRequest) => {
+  return NextResponse.json({});
+};
 
-  // simpler than webhooks, but constrained by serverless function timeout time
-  let isUploadSuccess = false;
-  let maxTries = 10;
-  let tries = 0;
-  while (!isUploadSuccess && tries < maxTries) {
-    const details = await fetch(
-      `https://livepeer.studio/api/asset/${requestedUrl.asset.id}`,
-      {
-        method: "GET",
-        headers: {
-          Authorization: `Bearer ${process.env.LIVEPEER_API_SECRET}`,
-          "Content-Type": "application/json",
-        },
-      }
-    );
-    const detailsJson = await details.json();
+export const GET = async (request: NextRequest) => {
+  let url = request.nextUrl.searchParams.get("url");
 
-    if (detailsJson.status !== "waiting") {
-      break;
-    }
+  // Exchange for livepeer url
+  const cid = url.replace("ipfs://", "");
+  const gatewayUrl = `${process.env.IPFS_DEFAULT_GATEWAY}/${cid}`;
```

Comment:
try: I think this needs to be "https://w3s.link/ipfs/${cid}" https://github.com/livepeer/livepeer-react/blob/main/packages/core-react/src/components/media/player/useSourceMimeTyped.tsx

Reply:
Tried it, still no caching

Reply:
ok will try emailing them again

Reply:
Interesting - it uploads fine without specifying a gateway, but still no caching (returns a different asset id each time it uploads)